[PR] Code refactor

LuchoTurtle opened the pull request:
closes #339 

**Can only be merged after #338 is merged.**

This splits the code into layers and adds export files.

Will add logging.

freitzzz commented:
Thanks for trusting my folder convention!

LuchoTurtle commented:
This should be ready to go. It's up to date with #338 , making it quickly mergeable after #338 is merged to `main`.

LuchoTurtle commented:
Going to focus on adding logging now, as initially stated.

freitzzz commented:
just saw that you guys are using `lumberdash` and that's awesome, it's my favourite logging package 😄

nelsonic commented on pubspec.yaml:
```diff
@@ -1,104 +1,32 @@
-# In Windows, build-name is used as the major, minor, and patch parts
-# of the product and file versions while build-number is used as the build suffix.
-version: 1.0.0+1
+name: dwyl_todo
```
In the _next_ PR, please begin renaming this to:
```suggestion
name: dwyl_app
```

LuchoTurtle replied:
Resolved in [`9482b94` (#340)](https://github.com/dwyl/app/pull/340/commits/9482b9453856033836d873e7b4e806b607d12573).

nelsonic reviewed (approved):
Looks good. Thanks @LuchoTurtle :shipit: 
@freitzzz thanks again for your expert eye. 🙏

nelsonic reviewed (dismissed):
Merging. :shipit:

nelsonic commented:
Merging blocked for some reason ...

![image](https://github.com/dwyl/app/assets/194400/92433e47-7e20-48a3-bbc2-a906d5a1b5e3)

nelsonic reviewed (approved):
@LuchoTurtle thanks for updating. 👌 
Sleep time! 😉

LuchoTurtle commented:
@nelsonic  I've updated the branch from `main` and there shouldn't be any merge conflicts.
I've additionally:
- changed the name to `dwyl_app`, as per your feedback.
- fixed the Node.js 12 deprecation notice-related issues in Github Actions.
